src/main/java/org/apache/pirk/query/wideskies/Query.java

```diff
@@ -26,41 +26,51 @@
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.function.Consumer;
 
+import com.google.gson.annotations.Expose;
 import org.apache.pirk.encryption.ModPowAbstraction;
 import org.apache.pirk.serialization.Storable;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * Class to hold the PIR query vectors
- *
  */
+
 public class Query implements Serializable, Storable
 {
-  private static final long serialVersionUID = 1L;
+  public static final long querySerialVersionUID = 1L;
+
+  // So that we can serialize the version number in gson.
+  @Expose public final long queryVersion = querySerialVersionUID;
 
   private static final Logger logger = LoggerFactory.getLogger(Query.class);
 
-  private final QueryInfo queryInfo; // holds all query info
+  @Expose private final QueryInfo queryInfo; // holds all query info
 
-  private final SortedMap<Integer,BigInteger> queryElements; // query elements - ordered on insertion
+  @Expose private final SortedMap<Integer,BigInteger> queryElements; // query elements - ordered on insertion
 
   // lookup table for exponentiation of query vectors - based on dataPartitionBitSize
   // element -> <power, element^power mod N^2>
   private Map<BigInteger,Map<Integer,BigInteger>> expTable = new ConcurrentHashMap<>();
 
   // File based lookup table for modular exponentiation
   // element hash -> filename containing it's <power, element^power mod N^2> modular exponentiations
-  private Map<Integer,String> expFileBasedLookup = new HashMap<>();
+  @Expose private Map<Integer,String> expFileBasedLookup = new HashMap<>();
```

Comment:
Should the hash -> filename map be serialized?
AFAIK it is only used by the responder to compute the results, so I don't think it needs serializing.